src/modules/back-to-top/back-to-top.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, ElementRef, Input, OnInit, ViewChild } from '@angular/core';
+
+@Component({
+  selector: 'stache-back-to-top',
+  templateUrl: './back-to-top.component.html',
+  styleUrls: ['./back-to-top.component.scss']
+})
+export class StacheBackToTopComponent {
+  @Input() offset: number;
+  @Input() color: string;
+  @Input() backgroundColor: string;
+  @ViewChild('backToTop') el: ElementRef;
+
+  ngOnInit(): void {
+    window.addEventListener('scroll', () => {
```

Comment:
It's dangerous to apply event listeners directly to the window in our components because we could potentially add multiple listeners in a given session. 

Look into implementing `@HostListener()`:
https://juristr.com/blog/2016/09/ng2-event-registration-document/